goal_src/engine/camera/cam-layout.gc

```diff
@@ -5,3 +5,5155 @@
+  (clmf-next-volume 0)
+  (set! *clm* *clm-edit*)
+  #t
+  )
+
+(define *last-cur-entity* -1)
+
+(defbehavior clmf-next-entity cam-layout ((arg0 int))
+  (let ((v1-0 (/ arg0 8)))
+   (when (zero? (-> self num-entities))
+    (set! (-> self cam-entity) #f)
+    (return #f)
+    )
+   (if (= (- v1-0) (-> self num-entities))
+    (set! v1-0 -1)
+    )
+   (if (= v1-0 (-> self num-entities))
+    (set! v1-0 1)
+    )
+   (set!
+    (-> self cur-entity)
+    (mod (+ v1-0 (-> self cur-entity)) (-> self num-entities))
+    )
+   )
+  (while (< (-> self cur-entity) 0)
+   (+! (-> self cur-entity) (-> self num-entities))
+   )
+  (set! *last-cur-entity* (-> self cur-entity))
+  (let ((v1-8 (-> self cur-entity))
+        (a0-13 (-> *camera-engine* alive-list next0))
+        )
+   *camera-engine*
+   (let ((a1-3 (-> a0-13 next0)))
+    (while (!= a0-13 (-> *camera-engine* alive-list-end))
+     (let ((a0-14 (-> (the-as connection a0-13) param1)))
+      (cond
+       ((zero? v1-8)
+        (set! (-> self cam-entity) (the-as entity-camera a0-14))
+        (set! *volume-descriptor-current* 0)
+        (set! *volume-point-current* 0)
+        (set! *volume-normal-current* 0)
+        (set! (-> self num-volumes) 0)
+        (cam-layout-entity-volume-info-create (-> self cam-entity) 'vol)
+        (set! (-> self first-pvol) (-> self num-volumes))
+        (cam-layout-entity-volume-info-create (-> self cam-entity) 'pvol)
+        (set! (-> self first-cutoutvol) (-> self num-volumes))
+        (cam-layout-entity-volume-info-create (-> self cam-entity) 'cutoutvol)
+        (set!
+         (-> *CAM_LAYOUT-bank* intro-step)
+         (cam-slave-get-intro-step (-> self cam-entity))
+         )
+        (return #f)
+        )
+       (else
+        (+! v1-8 -1)
+        )
+       )
+      )
+     (set! a0-13 a1-3)
+     *camera-engine*
+     (set! a1-3 (-> a1-3 next0))
+     )
+    )
+   )
+  #t
+  )
+
+(defbehavior clmf-to-vol-attr cam-layout ()
+  (set! (-> self res-key) (the float (-> self cur-volume)))
+  (set! *clm* *clm-vol-attr*)
+  #t
+  )
+
+(defun clmf-to-spline-attr ()
+  (set! *clm* *clm-spline-attr*)
+  #t
+  )
+
+(defun clmf-to-intro-attr ()
+  (set! *clm* *clm-intro-attr*)
+  #t
+  )
+
+(defun clmf-to-index-attr ()
+  (set! *clm* *clm-index-attr*)
+  #t
+  )
+
+(defun clmf-to-focalpull-attr ()
+  (set! *clm* *clm-focalpull-attr*)
+  #t
+  )
+
+(defbehavior clmf-to-edit cam-layout ()
+  (set! (-> self res-key) (-> (new 'static 'array float 1 -1000000000.0) 0))
+  (set! *clm* *clm-edit*)
+  #t
+  )
+
+(defun clmf-to-select ()
+  (set! *camera-layout-blink* #f)
+  (set! *clm* *clm-select*)
+  #t
+  )
+
+(defbehavior clmf-look-through cam-layout ()
+  (set! (-> *camera-other-fov* data) (cam-slave-get-fov (-> self cam-entity)))
+  (cam-slave-get-vector-with-offset
+   (the-as entity-actor (-> self cam-entity))
+   *camera-other-trans*
+   'trans
+   )
+  (set! (-> *camera-other-root* quad) (-> *camera-other-trans* quad))
+  (cam-slave-get-rot
```

Comment:
cam-slave-get-rot should probably take a `matrix` instead of a `quaternion` as the second arg